c_common/models/live_packet_gather/live_packet_gather.c

```diff
@@ -111,19 +122,19 @@ void flush_events(void) {
             // insert appropriate header
             sdp_msg_aer_header[0] = 0;
             sdp_msg_aer_header[0] |= temp_header;
-            sdp_msg_aer_header[0] |= (event_count & 0xff);
+            sdp_msg_aer_header[0] |= CLAMP8(event_count);
 
             g_event_message.length = sizeof(sdp_hdr_t) + header_len
                                      + event_count * event_size;
 
             if (payload_apply_prefix && payload_timestamp) {
                 uint16_t *temp = (uint16_t *) sdp_msg_aer_payload_prefix;
 
-                if (!(packet_type && 0x2)) {
```

Comment:
On the naming of the EIEIO header bits, heres the doc that covers it. 

http://spinnakermanchester.github.io/2015.004.LittleRascal/EIEIOProtocolProposal.pdf

0x2 is F (or packet format (command or basic)
0x1 is P (with or without key prefix, which is its attempt of compression)

Hope that helps. That whole code and its corrasponding RIPMCS needs a overall in all honestly.